documentation/api/io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListener.adoc

```diff
@@ -213,9 +213,109 @@ spec:
           preferredNodePortAddressType: InternalDNS
     #...
 ----
-+
 NOTE: TLS hostname verification is not currently supported when exposing Kafka clusters using node ports.
 
+cluster-ip:: Configures an external listener to expose Kafka `cluserIP` type `Services`.
++
+This listener type can be helpful if you:
+
+* don't want to manage dynamic broker adresses with nodeport services
+* don't want to bring new LoadBalancer and external IP by broker (+ bootstrap)
+* can't use tls in your existing kafka clients
+* can't enable ssl passthrough in your ingress controller
++
+
++
+Using only one LoadBalancer, brokers could be exposed using ingress controller.
++
+A new clusterIP service is created for each Kafka broker pod.
++
+An additional clusterIP is created to serve as a Kafka _bootstrap_ address.
++
+The client connects on the `advertisedPort`, 9190 for broker 0, on `advertisedHost`, ingress.mycompany.com in the following example, but traffic is then routed to the port you configure,
+which is `9096` in the following example.
++
+`advertisedHost` ingress.mycompany.com should be a dns name routed to ingress LoadBalancer service. In the Ingress Controller LoadBalancer service,
+we should add broker exposed ports + extra bootstrap port.
+
+.Example `cluster-ip` listener configuration
+[source,yaml,subs="+attributes"]
+----
+#...
+spec:
+  kafka:
+    #...
+    listeners:
+      - name: external-cluster-ip
+        type: cluster-ip
+        tls: false
+        port: 9096
+        configuration:
+          brokers:
+            - broker: 0
+              advertisedHost: ingress.mycompany.com
+              advertisedPort: 9190
+            - broker: 1
+              advertisedHost: ingress.mycompany.com
+              advertisedPort: 9191
+            - broker: 2
+              advertisedHost: ingress.mycompany.com
+              advertisedPort: 9192
```

Comment:
One concern about this API is it makes it harder to scale the cluster up, since you have to update this configuration with new brokers prior to the actual scale up. But then you also have to update the Service and the ConfigMap, so perhaps it's inevitable that this can't really work easily with scale up.

Reply:
sure, since the ingress exposition are a manual operation, it will be hard to have a scalable solution. We still need manuel and hard configuration to make it working